pkg/actions/kv_run_results_iterator_test.go

```diff
@@ -213,12 +223,23 @@ func createTestData(t *testing.T, ctx context.Context, kvStore kv.StoreMessage)
 	badValue := "BadValue"
 	key := "aaa"
 	require.NoError(t, kvStore.Set(ctx, []byte(actions.PartitionKey), []byte(key), []byte(badValue)))
-	msgIdx++
 	keyMap[key] = msgIdx
-	keyList[msgIdx] = key
+	keyList = append(keyList, key)
 
 	require.NoError(t, kvStore.Set(ctx, []byte(actions.PartitionKey), []byte(IndexOutOfRange), []byte(badValue)))
+	key = "zzz"
 	msgIdx++
 	keyMap[key] = msgIdx
-	keyList[msgIdx] = key
+	keyList = append(keyList, key)
+
+	// reverse the list to make it sorted in ascending order
+	reverse(keyList)
+
+	return keyMap, keyList
+}
+
+func reverse(s []string) {
```

Comment:
You know the size of the list, you can create an array and insert the items in reverse order, I think this func is redundant

Reply:
It is, but with all the mappings it makes handling `keyList` & `keyMap` so much harder that I wish to avoid another off-by-one mistake. Reversing is much easier than inserting in reverse order.